### PR TITLE
Feature/issue 143 parse game code

### LINKED ISSUE #143: parse game code from URL
To avoid users being required to strip the game code out or add it manually to the URL just always pass URLs and handle stripping the URL as part of the input validation

### PR DESCRIPTION
**Pull request by creisle**
closes #143

### CODE REVIEW

File: src/components/StartGameDialog/index.tsx [hunk 1]
```diff
@@ -45,7 +54,7 @@ function StartGameDialog(props: StartGameDialogProps) {
             <DialogContent>
                 {isJoining && (<TextField
                     onChange={handleGameIdChange}
-                    label="enter a game ID"
+                    label="Enter a game URL or ID"
```
Comment (areisle) [inline src/components/StartGameDialog/index.tsx]: can this same 'URL or game ID' instead? I think it sounds less confusing

File: src/components/Footer/index.tsx [hunk 2]
```diff
@@ -26,7 +26,7 @@ function Footer(props: FooterProps) {
 
     const handleAddToClipboard = () => {
         if (!gameCode) { return; }
-        copyToClipboard(gameCode);
+        copyToClipboard(`${window.location.origin}${window.location.pathname}?game=${gameCode}`);
```
Comment (areisle) [inline src/components/Footer/index.tsx]: I wonder if you could just use location.href here

File: src/components/StartGameDialog/index.tsx [hunk 3]
```diff
@@ -17,15 +17,24 @@ interface StartGameDialogProps {
...
         onStart,
     } = props;
 
     const [gameId, setGameId] = useState('');
 
     const handleGameIdChange: TextFieldProps['onChange'] = (event) => {
-        setGameId(event?.target?.value);
+        let gameCode = event?.target?.value;
+        if (gameCode.includes('?')) {
+            gameCode.split('?')[1].split('&').forEach((param) => {
```
Comment (areisle) [inline src/components/StartGameDialog/index.tsx]: I would do 
```javascript
url = new URL(gameCode)
gameCode = url.searchParams.get('game')
```
instead

File: src/components/StartGameDialog/index.tsx (hunk 1 above)
Comment (creisle) [reply src/components/StartGameDialog/index.tsx]: sure

File: src/components/StartGameDialog/index.tsx (hunk 3 above)
Comment (creisle) [reply src/components/StartGameDialog/index.tsx]: yup that works

File: src/components/Footer/index.tsx (hunk 2 above)
Comment (creisle) [reply src/components/Footer/index.tsx]: I didn't because location.href includes the search params

Comment (areisle) [reply src/components/Footer/index.tsx]: but the only thing in the query string is the game code which is what we want

Comment (creisle) [reply src/components/Footer/index.tsx]: hmm ya i can't remember why i thought i needed it separate.

**Review by areisle — APPROVED** [review]

**Pull request merged**